### ISSUE TITLE
[BUG] Setting a variable to an inventory type intending to use as custom inventory fails

### ISSUE THREAD

**Issue by Fusezion** [SkriptLang/Skript#5925]
### Skript/Server Version

```markdown
[22:16:08 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[22:16:08 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[22:16:08 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[22:16:08 INFO]: [Skript] Server Version: git-Paper-126 (MC: 1.20.1)
[22:16:08 INFO]: [Skript] Skript Version: 2.8.0-dev (selfbuilt-unknown)
[22:16:08 INFO]: [Skript] Installed Skript Addons:
[22:16:08 INFO]: [Skript]  - skript-reflect v2.3.1-for-skript-beta (https://github.com/TPGamesNL/skript-reflect)
[22:16:08 INFO]: [Skript]  - SkBee v2.16.1 (https://github.com/ShaneBeee/SkBee)
[22:16:08 INFO]: [Skript] Installed dependencies:
[22:16:08 INFO]: [Skript]  - Vault v1.7.3-b131
```

### Bug Description

When you do `set {_inv} to chest inventory` it's saved as a inventory type, skript fails to tell it should be a custom inventory and when you attempting to do `set slot x of {_inv} to y` it just sightly fails as it's not a custom inventory but an inventory type

### Expected Behavior

Skript to understand `set {_inv} to chest inventory` might be expecting to be used as a custom inventory and not an inventory type

### Steps to Reproduce

`{_inv}` will never have an item set however when you open `{_inv2}` the first slot will be stone
```
command /example:
  trigger:
    set {_inv} to chest inventory
    set {_inv2} to chest inventory named ""
    set slot 0 of {_inv} and {_inv2} to stone
    broadcast "inv 1"
    open {_inv} to player
    wait a second
    broadcast "inv 2"
    open {_inv2} to player
```

### Errors or Screenshots

![image](https://github.com/SkriptLang/Skript/assets/81952476/4b75730d-ca4d-44ee-aca4-2c8c57567eb9)

### Other

The only solution I could see is always making plain `chest inventory` refer to a custom inventory and rely on some other method for inventory type comparisons

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by sovdeeth** [comment]
Replicated, marking as medium due to prevalence of vanille gui use. Arguable something that needs to be fixed before 2.7 releases.

**Comment by APickledWalrus** [comment]
is this actually a bug though? hasn't this always been the behavior?

**Comment by sovdeeth** [comment]
I'm unsure about the class stuff, but the part i replicated was being unable to set slots in the inventory

**Comment by Fusezion** [comment]
There's a chance it might actually of been but the way `%inventorytype% named %string%` is handled with properly making it a custom inventory feels weird that just a plain inventory type alone  with no name doesn't support it.

Honestly I'm unsure where skript even has `%inventorytype% named %string%` as `ExprChestInventory` only does chest meaning it should be done via `ExprNamed` which doesn't make sense to me

I don't recall 2.6.4 treating `set [_inv} to chest inventory` like how it does now tho

**Comment by sovdeeth** [comment]
It appears I was indeed mistaken, this happens in 2.6.4 as well, so I guess it isn't as pressing of an issue as I believed.

**Comment by Fusezion** [comment]
little wordy rename any better one I'm welcome for

**Comment by TheLimeGlass** [comment]
I do not agree with this change. Skript is working as intended. `chest inventory` should reflect an InventoryType to me.

If you changed ExprChestInventory to not require either a name or rows definition, you would have conflict issues between ExprChestInventory and the actual InventoryType ClassInfo. Skript would pick the ClassInfo in that case, so it still wouldn't work unless you added `[new]` in the user statement `set {_inventory} to a new chest inventory`

**Comment by Fusezion** [comment]
If we aren't going to do this then making a different syntax to do so would be the next best option in my opinion.
I don't believe it's good to rely on `<inventoryType> <ExprName>` for making a custom inventory. 
I'd prefer to see syntax like below where you know what's doing what and how it's occurring.

```applescript
[a] [new] custom %inventorytype% [(named|with name) %-string%]
[a] [new] chest inventory (named|with name) %string% [with %-number% row[s]]
[a] [new] chest inventory with %number% row[s] [(named|with name) %-string%]
# I would add an optional [custom] to chest syntax to keep it in line with pattern 0
``` 

would this be a breaking change? eh depends, if we remove ExprName's inventorytype support like I'd prefer yes but knowing the team here you probably wouldn't and instead this would just be an enhancement for something that makes better sense than `ExprCustomChestInventory`.

I personally dislike `custom` being forced however if I remember skript also dislikes using `new` to define a change in something, as such `custom` is the only other safe bet

example syntax in use would be
```applescript
set {_shulker} to custom shulker box inventory
set {_loom} to a new custom loom inventory named "Let's Get Looming"
set {_chest} to chest inventory named "blah"
```
